src/screens/Profiles/UserLibrary/components/UserLibraryListCard/component.js

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
+import { debounce } from 'lodash';
+import { PropTypes } from 'prop-types';
+import { Image, Text, TouchableHighlight, View } from 'react-native';
+import { Counter } from 'kitsu/components/Counter';
+import { ProgressBar } from 'kitsu/components/ProgressBar';
+import { Rating } from 'kitsu/components/Rating';
+import { SelectMenu } from 'kitsu/components/SelectMenu';
+import Swipeable from 'react-native-swipeable';
+import { styles } from './styles';
+
+const menuImage = require('kitsu/assets/img/menus/three-dot-horizontal-grey.png');
+
+const STATUS_SELECT_OPTIONS = [
+  { value: 'current', anime: 'Currently Watching', manga: 'Currently Reading' },
+  { value: 'planned', anime: 'Want To Watch', manga: 'Want To Read' },
+  { value: 'onHold', anime: 'On Hold', manga: 'On Hold' },
+  { value: 'dropped', anime: 'Dropped', manga: 'Dropped' },
+  { value: 'completed', anime: 'Completed', manga: 'Completed' },
+  { value: 'remove', anime: 'Remove From Library', manga: 'Remove From Library' },
+  { value: 'cancel', anime: 'Nevermind', manga: 'Nevermind' },
+];
+
+export class UserLibraryListCard extends React.Component {
+  static propTypes = {
+    currentUser: PropTypes.object.isRequired,
+    data: PropTypes.object.isRequired,
+    libraryStatus: PropTypes.string.isRequired,
+    libraryType: PropTypes.string.isRequired,
+    profile: PropTypes.object.isRequired,
+    updateUserLibraryEntry: PropTypes.func.isRequired,
+  }
+
+  state = {
+    libraryStatus: this.props.data.status,
+    progress: this.props.data.progress,
+    progressPercentage: Math.floor((this.props.data.progress / this.getMaxProgress()) * 100),
+    ratingTwenty: this.props.data.ratingTwenty,
+    sliderCanActivate: false,
+  }
+
+  onProgressValueChanged = (newProgress) => {
+    const maxProgress = this.getMaxProgress();
+    const progressPercentage = Math.floor((newProgress / maxProgress) * 100);
+    this.setState({
+      progressPercentage,
+      progress: newProgress,
+    }, this.debounceSave);
+  }
+
+  onRatingChanged = (ratingTwenty) => {
+    this.setState({ ratingTwenty }, this.debounceSave);
+  }
+
+  onRightButtonsActivate = () => {
+    this.setState({ sliderCanActivate: true });
+  }
+
+  onRightButtonsDeactivate = () => {
+    this.setState({ sliderCanActivate: false });
+  }
+
+  onStatusSelected = (libraryStatus) => {
+    this.setState({ libraryStatus }, this.debounceSave);
+  }
+
+  getMaxProgress() {
+    const { data, libraryType } = this.props;
+    const mediaData = data[libraryType];
+
+    if (mediaData.type === 'anime') {
+      return mediaData.episodeCount;
+    }
+
+    return mediaData.chapterCount;
+  }
+
+  saveEntry = () => {
+    // send the status from props because that is the list we're looking
```

Comment:
Would be good to disable editing while you're making the API requests. I notice they're debounced, but you still might have the scenario where it takes more than 100ms to do the action and you'd get multiple actions hitting the server. That might be ok in this case, so ignore me if that's not a problem, just wanted to make sure you knew it's a possibility.